engine/crates/composition/src/subgraphs/directives.rs

```diff
@@ -248,6 +257,15 @@ pub(crate) struct OverrideDirective {
     pub(crate) label: Option<StringId>,
 }
 
+#[derive(Debug)]
+pub(crate) struct AuthorizedDirective {
+    pub(crate) rule: StringId,
+    pub(crate) arguments: Option<Vec<Selection>>,
+    pub(crate) fields: Option<Vec<Selection>>,
+    /// We expect JSON here, but since this will be passed as-is, we don't need to parse.
```

Comment:
well by `JSON` I meant the JSON scalar where anything works, so should be of type `Value` like:
```
@authorized(metadata: { a: 1 })
```
We also can't rely here on the format we might use to send data to hooks.

Reply:
The policy example uses this for example:
```
type User {
  id: ID!
  contracts: Contract @authorized(rule: "policy", metadata: [["role:admin"]] )
}
```

Reply:
Aah ok, yeah, that needs to be changed then. I assumed a custom scalar with JSON in a strin.